### ISSUE TITLE
Fails to generate correct flow types when field and getter methods are inconsistently named

### ISSUE THREAD

**Issue by havardh** [havardh/javaflow#40]
Javaflow generates flow types based on the naming of the field. Json libraries like the one in Jackson generates the json payload based on getters.

We should either change javaflow to look at the getter names or keep it simple and throw an error/warning when inconsistencies are found.

**Comment by havardh** [comment]
To solve this with an error/warning I suggest that we add an alternative to the `ClassVisitor` (`
extends VoidVisitorAdapter<...>`) which is responsible for validating the class. 

This can be called from here [JavaParser.java#L40](https://github.com/havardh/javaflow/blob/master/src/main/java/no/havard/javaflow/phases/parser/java/JavaParser.java#L40) like:

```
ClassGetterNamingValidator validator = new ClassGetterNamingValidator();
visitor.visit(cu, validator);
```

It should visit all fields and all methods and build up a new validator object which can then be queried for the validation. 

The validator object would be something like this: 

```
class ClassGetterNamingValidator {
  List<String> fields;
  List<String> getters;

  addField(String);
  addGetterMethod(String)
  validate() {
    // validate the fields and getters
  }
}
```

This should be used inside the vistor like

```
public void visit(FieldDeclaration field, ClassGetterNamingValidator validator) {
  super.visit(field, validator);
  field.getVariables().forEach(variable -> validator.addField(variable.getId().getName());
}

public void visit(MethodDeclaration method, ClassGetterNamingValidator validator) {
  super.visit(method, arg);
  if (isGetter(method)) {
    validator.addGetterMethod(method.getName()
  }
}
```

Now we can use the `validator.isValid()` method to check if we need to throw an error/warning in [JavaParser.java#L40](https://github.com/havardh/javaflow/blob/master/src/main/java/no/havard/javaflow/phases/parser/java/JavaParser.java#L40).

**Comment by havardh** [comment]
Fixed by #45
